core/input/input_event.h

```diff
@@ -292,15 +292,15 @@ class InputEventMouseMotion : public InputEventMouse {
 
 class InputEventJoypadMotion : public InputEvent {
 	GDCLASS(InputEventJoypadMotion, InputEvent);
-	int axis = 0; ///< Joypad axis
+	JoyAxis axis = (JoyAxis)0; ///< Joypad axis
```

Comment:
Shouldn't it be `JOY_AXIS_INVALID` (-1) by default?

Reply:
Maybe. That would change behavior beyond the scope of this PR, so it can be done in another PR if desired.

Reply:
Yeah, agree it's out of the scope of this PR.
But shouldn't then initializations for `MouseButton` and `MIDIMessage` be analogous? Like `MouseButton button_index = (MouseButton)0;` instead of `MouseButton button_index = MOUSE_BUTTON_NONE;`.

Reply:
`JoyButton` and `JoyAxis` already have well-defined members with `0`, so we can't have `JOY_BUTTON_NONE` or `JOY_AXIS_NONE` (unless we want to give it some weird value like -2). We could use `(MouseButton)0`, but I think `MOUSE_BUTTON_NONE` is nicer. Semantically speaking, `(JoyAxis)0` means "I don't care what this should be, just initialize it to the zero member", but kinda assumes that `0` is already a member of the enum, we just don't care which.